apps/vs-code-designer/src/app/commands/dataMapper/FxWorkflowRuntime.ts

```diff
@@ -89,60 +81,39 @@ async function createJsonFile(
 }
 
 async function waitForBackendRuntimeStartUp(url: string, initialTime: number): Promise<void> {
-  while (!(await isBackendRuntimeUp(url)) && new Date().getTime() - initialTime < dataMapLoadTimeout) {
+  while (!(await isDesignTimeUp(url)) && new Date().getTime() - initialTime < dataMapLoadTimeout) {
     await delay(1000); // Re-poll every X ms
   }
 
-  if (await isBackendRuntimeUp(url)) {
+  if (await isDesignTimeUp(url)) {
     return Promise.resolve();
   } else {
     return Promise.reject();
   }
 }
 
-async function delay(ms: number): Promise<void> {
-  return new Promise((resolve) => setTimeout(resolve, ms));
-}
-
-async function isBackendRuntimeUp(url: string): Promise<boolean> {
-  try {
-    await fetch(url);
-    return Promise.resolve(true);
-  } catch (ex) {
-    return Promise.resolve(false);
-  }
-}
-
 function startBackendRuntimeProcess(workingDirectory: string | undefined, command: string, ...args: string[]): void {
```

Comment:
This one seems to contain quite a bit of duplicates to `startDesignTimeProcess` in `startDesignTimeApi.ts`, might be worth coming back to it later to reuse the function!

Reply:
True, I will update the code to reuse it. Thanks!